src/Features/LanguageServer/Protocol/Handler/ExportLspMethodAttribute.cs

```diff
@@ -15,9 +17,11 @@ internal class ExportLspMethodAttribute : ExportAttribute, IRequestHandlerMetada
     {
         public string MethodName { get; }
 
-        public string LanguageName { get; }
+        public string? LanguageName { get; }
+
+        public bool MutatesSolutionState { get; }
 
-        public ExportLspMethodAttribute(string methodName, string languageName = null) : base(typeof(IRequestHandler))
+        public ExportLspMethodAttribute(string methodName, string? languageName = null, bool mutatesSolutionState = false) : base(typeof(IRequestHandler))
```

Comment:
i would not make this optional.  all features should have to think about this and be explicit and self documenting.

Reply:
Agreed, will be in a future PR

Reply:
hrmm... can it be non-optional *in this PR*?